src/Microsoft.Diagnostics.Monitoring.Options/ExtensionEgressProviderOptions.cs

```diff
@@ -5,15 +5,14 @@
 using Microsoft.Diagnostics.Monitoring.WebApi;
 using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
-using System.Text.Json.Serialization;
 
 namespace Microsoft.Diagnostics.Tools.Monitor.Egress
 {
     /// <summary>
     /// Egress provider options for external egress providers.
     /// </summary>
     internal sealed partial class ExtensionEgressProviderOptions : 
-        Dictionary<string, string>,
+        Dictionary<string, object>,
```

Comment:
This isn't implemented in this PR, but in the initial iteration of this design we were assuming that we could treat all of our extension's configuration as a `Dictionary<string, string>`. The addition of the Azure provider's `Metadata` (used for custom metadata) breaks this assumption, and will require changes to how we pass the configuration to the extension. To simplify this PR, I am not currently passing the `Metadata` section of configuration, but this will be fixed in a future PR.